ambry-router/src/main/java/com.github.ambry.router/GetBlobInfoOperation.java

```diff
@@ -298,21 +347,39 @@ void onErrorResponse(ReplicaId replicaId) {
   }
 
   /**
-   * Handle the body of the response: Deserialize and set the {@link BlobInfo} to return.
+   * Handle the body of the response: Deserialize and set the {@link BlobInfo} to return if no decryption is required.
+   * If decryption is required, submit a jon for decryption.
    * @param payload the body of the response.
    * @param messageMetadata the {@link MessageMetadata} associated with the message.
    * @throws IOException if there is an IOException while deserializing the body.
    * @throws MessageFormatException if there is a MessageFormatException while deserializing the body.
    */
   private void handleBody(InputStream payload, MessageMetadata messageMetadata)
       throws IOException, MessageFormatException {
-    ByteBuffer encryptionKey = messageMetadata == null ? null : messageMetadata.getEncryptionKey();
-    // @todo use the encryption key for decryption.
-    if (operationResult == null) {
-      BlobInfo blobInfo = new BlobInfo(MessageFormatRecord.deserializeBlobProperties(payload),
-          MessageFormatRecord.deserializeUserMetadata(payload).array());
-      getOptions().ageAtAccessTracker.trackAgeAtAccess(blobInfo.getBlobProperties().getCreationTimeInMs());
-      operationResult = new GetBlobResultInternal(new GetBlobResult(blobInfo, null), null);
+    if (!successfullyDeserialized) {
```

Comment:
Trying to understand this interaction
1) Response is received from one server
2) The response is deserialized but needs decryption
3) Decryption fails so we need to ask other servers for the data
4) This piece of code can never execute again because `successfullyDeserialized` is never reset?

What is this check here for btw?

Reply:
As I quoted earlier, we don't accept responses from more servers, if we were able to successfully deserialize response from one.